plugins/inputs/mongodb/mongodb.go

```diff
@@ -164,6 +168,24 @@ func (m *MongoDB) Gather(acc telegraf.Accumulator) error {
 		wg.Add(1)
 		go func(srv *Server) {
 			defer wg.Done()
+			if !srv.reachable {
+				// is not reachable try a reconnect
+				disconnectCtx, disconnectCancel := context.WithTimeout(context.Background(), 1*time.Second)
+				defer disconnectCancel()
+				err := srv.client.Disconnect(disconnectCtx)
+				if err != nil {
+					m.Log.Errorf("unable to reconnect to MongoDB: %q", err)
+					return
```

Comment:
Because we aren't returning an error, this would cause a retry to happen everytime `Gather` is called right? This could cause a lot of "unable to reconnect to mongoDB" log messages to show up. Maybe adding a configuration option for the user to set how often to retry would help? Or how many times to retry before either exiting or only stop retrying for this instance?

Reply:
Yes totally, this the first behavior we wanted on our side (as a draft I did not invest to much time on it as I wanted to be shure it will have an interest for Telegraf upstream).